src/Custom/RealtimeConversation/Internal/AsyncWebsocketMessageEnumerator.cs

```diff
@@ -26,8 +27,12 @@ public AsyncWebsocketMessageResultEnumerator(WebSocket webSocket, CancellationTo
 
     public ValueTask DisposeAsync()
     {
-        ArrayPool<byte>.Shared.Return(_receiveBuffer);
-        _webSocket?.Dispose();
+        if (!_disposed)
```

Comment:
this is not thread safe.

Reply:
Yes, but do we expect a given instance of the enumerator to be run concurrently?

Reply:
probably not in real scenarios, but double return to a pool is such a severe bug. I think we should not be using the pool unless we can guarantee 100% correctness, even in contrived or bug cases.

Reply:
Wouldn't hurt to use an Interlocked, e.g.
```C#
if (Interlocked.Exchange(ref _receiveBuffer, null) is byte[] toReturn)
{
    ArrayPool<byte>.Shared.Return(toReturn);
}
```

Reply:
tru